### PR TITLE
Resolved the bug The banner message is causing the search bar to be o…

### LINKED ISSUE #20608: [Bug Report] The banner message is causing the search bar to be obscured.
**Expected behavior and actual behavior:**

- Expected behavior: After searching, if there is a banner message at the top of the page, it `will not` causing the search bar to be obscured.

- Actual behavior: After searching, if there is a banner message at the top of the page, it `will` causing the search bar to be obscured.
image

```bash
body > harbor-app > harbor-shell > clr-main-container > div > div > search-result > div
```

**Steps to reproduce the problem:**
Search anything when there's a `banner message` on the top
reproduce video: 

https://github.com/BrettDean/temp/raw/main/20240624/output.mp4

<video controls>
  <source src="https://github.com/BrettDean/temp/raw/main/20240624/output.mp4" type="video/mp4">
  Your browser does not support the video tag.
</video>

This issue can be reproduced on macOS Sonoma 14.5 in the following browsers:

- Safari 17.5 (19618.2.12.11.6)
- Firefox 127.0
- Chrome 125.0.6422.176

**Versions:**

- harbor version: v2.11.0
- docker engine version: Docker version 26.1.4, build 5650f9b
- docker-compose version: Docker Compose version v2.27.0

### PR DESCRIPTION
**Pull request by kunal-511**
…bscured #20608

Thank you for contributing to Harbor!

# Comprehensive Summary of your change
Bug Report: The banner message was obscuring the search bar. #20608
Resolution: The issue has been resolved by adjusting some styles to ensure the below content does not overlap the search bar.
Details: After applying the style changes, the search bar is now fully visible even with the banner message displayed at the top of the page.
For the proof attaching the screenshot 
image

# Issue being fixed
Fixes #20608 

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Comment by kunal-511** [comment]
Please review this PR @zyyw

**Comment by AllForNothing** [comment]
@ kunal-511 Can you use a dynamic top value to fit the different cases? Using the document.query function to know if there is a banner
1.  has banner:   top  = 95px
2.  no banner:    top = 60px

**Comment by kunal-511** [comment]
Okay I will look into this

**Comment by kunal-511** [comment]
@AllForNothing I have added a new commit which now dynamically update the top value depending on wether the banner message is there or not

**Comment by AllForNothing** [comment]
@kunal-511 Can you refer to this https://github.com/AllForNothing/harbor/tree/top-value-search,  then update your PR?
the hasValidBannerMessage function  is inaccurate as Harbor may have a banner in other cases

**Comment by AllForNothing** [comment]
@kunal-511 Can you refer to this https://github.com/AllForNothing/harbor/tree/top-value-search,  then update your PR?
the hasValidBannerMessage function  is inaccurate as Harbor may have banner in other cases

**Comment by kunal-511** [comment]
@AllForNothing I have updated the logic of top value by referring to your feedback. Please check it.

> @kunal-511 Can you refer to this [main...AllForNothing:harbor:top-value-search](https://github.com/goharbor/harbor/compare/main...AllForNothing:harbor:top-value-search), then update your PR? the hasValidBannerMessage function is inaccurate as Harbor may have a banner in other cases

File: src/portal/src/app/shared/components/global-search/search-result.component.ts [hunk 1]
```diff
@@ -169,11 +176,27 @@ export class SearchResultComponent implements OnInit, OnDestroy {
...
             }
         );
     }
+    getTopValue(): number {
+        const headerHeight: number = document.querySelector('navigator')?.clientHeight || 10;
+        const bannerHeight: number = document.querySelector('app-app-level-alerts')?.clientHeight || 10;
+        return headerHeight + bannerHeight;
+    }
+
+    private adjustOverlayTop(): void {
```
Comment (AllForNothing) [inline src/portal/src/app/shared/components/global-search/search-result.component.ts:193]: if you use this function to re-render the dom, then you can remove the style binding in `[search-result.component.html](https://github.com/goharbor/harbor/pull/20794/files#diff-ec327ae356d30c4098c563bfce30fa1e746dfc33c449e9dcdd0a273c11297c7c)`

File: src/portal/src/app/shared/components/global-search/search-result.component.ts [hunk 2]
```diff
@@ -169,11 +176,27 @@ export class SearchResultComponent implements OnInit, OnDestroy {
...
             },
             error => {
                 this.onGoing = false;
                 this.msgHandler.handleError(error);
             }
         );
     }
+    getTopValue(): number {
+        const headerHeight: number = document.querySelector('navigator')?.clientHeight || 10;
+        const bannerHeight: number = document.querySelector('app-app-level-alerts')?.clientHeight || 10;
```
Comment (AllForNothing) [inline src/portal/src/app/shared/components/global-search/search-result.component.ts:189]: Why 10  not 0 ?

**Comment by kunal-511** [comment]
@AllForNothing  I have added a new commit please check

**Comment by AllForNothing** [comment]
@kunal-511 What about just cherrying-pick the changes from https://github.com/goharbor/harbor/compare/main...AllForNothing:harbor:top-value-search.

BTW,  you need to sign your commits with 'git commit -s' to pass the DCO check
and run 'npm run lint_fix' to format your code

**Comment by kunal-511** [comment]
@AllForNothing  I have added a new commit. Have a look at it .

File: src/portal/src/app/shared/components/global-search/search-result.component.ts (hunk 1 above)
Comment (AllForNothing) [reply src/portal/src/app/shared/components/global-search/search-result.component.ts:201]: Please remove this function

File: src/portal/src/app/shared/components/global-search/search-result.component.ts [hunk 3]
```diff
@@ -169,11 +182,27 @@ export class SearchResultComponent implements OnInit, OnDestroy {
...
+            document.querySelector('navigator')?.clientHeight || 0;
+        const bannerHeight: number =
+            document.querySelector('app-app-level-alerts')?.clientHeight || 0;
+        return headerHeight + bannerHeight;
+    }
+
+    private adjustOverlayTop(): void {
+        const topValue = this.getTopValue();
+    }
+    @HostListener('window:resize', ['$event'])
```
Comment (AllForNothing) [inline src/portal/src/app/shared/components/global-search/search-result.component.ts:204]: Please remove this listener

File: src/portal/src/app/shared/components/global-search/search-result.component.ts [hunk 4]
```diff
@@ -50,7 +58,10 @@ export class SearchResultComponent implements OnInit, OnDestroy {
     constructor(
         private search: GlobalSearchService,
         private msgHandler: MessageHandlerService,
-        private searchTrigger: SearchTriggerService
+        private searchTrigger: SearchTriggerService,
+        private renderer: Renderer2,
+        private el: ElementRef,
+        private appConfigService: AppConfigService
```
Comment (AllForNothing) [inline src/portal/src/app/shared/components/global-search/search-result.component.ts:64]: Please remove these services

**Pull request closed without merging**